### PR TITLE
Overhaul libssl error handling part 2

### PR DESCRIPTION
**Pull request by mattcaswell**
This PR continues where #4778 left off. This time it focuses on changing the record layer to use SSLfatal().

### CODE REVIEW

File: ssl/record/rec_layer_s3.c [hunk 1]
```diff
@@ -506,6 +516,7 @@ int ssl3_write_bytes(SSL *s, int type, const void *buf_, size_t len,
 
             i = ssl3_write_pending(s, type, &buf[tot], nw, &tmpwrit);
             if (i <= 0) {
+                /* SSLfatal() already called if appripriate */
```
Comment (FdaSilvaYY) [inline ssl/record/rec_layer_s3.c]: s/appripriate /appropriate

Comment (mattcaswell) [reply ssl/record/rec_layer_s3.c]: Fixed.

**Comment by mattcaswell** [comment]
Rebased to resolve some merge conflicts.

Still looking for a reviewer for this.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**